db/compaction_job.cc

```diff
@@ -892,7 +893,9 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       &existing_snapshots_, earliest_write_conflict_snapshot_,
       snapshot_checker_, env_, ShouldReportDetailedTime(env_, stats_), false,
       &range_del_agg, sub_compact->compaction, compaction_filter,
-      shutting_down_, preserve_deletes_seqnum_));
+      shutting_down_, preserve_deletes_seqnum_,
+      // Currently range_del_agg is incompatible with snapshot refresh feature.
+      range_del_agg.IsEmpty() ? snap_list_callback_ : nullptr));
```

Comment:
We have a callback if range_del_agg is empty, but no callbck if it is not empty. It's not intuitive to me.

Reply:
Is this sufficient to disable the snapshot refresh mechanism in the presence of range tombstones? `RangeDelAggregator` is populated dynamically as sstables are iterated over during compaction. I would guess that `range_del_agg.IsEmpty()` could return true at the start of a compaction and false later on when an sstable with range tombstones is encountered.

Reply:
@petermattis you are correct. That's why @maysamyabandeh added comment to the option that it isn't compatible with range delete: https://github.com/facebook/rocksdb/pull/5625

btw, we might have found it isn't compatible with another feature recently.